sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultEnvironmentPostProcessor.java

```diff
@@ -171,38 +154,60 @@ SecretClient buildSecretClient(AzureKeyVaultSecretProperties secretProperties) {
         return new SecretClientBuilderFactory(secretProperties).build().buildClient();
     }
 
-    private AzureKeyVaultSecretProperties loadProperties(Binder binder) {
-        AzureGlobalProperties azureProperties = binder
+    AzureKeyVaultSecretProperties loadProperties(ConfigurableEnvironment environment) {
+        Binder binder = Binder.get(environment);
+        AzureGlobalProperties globalProperties = binder
             .bind(AzureGlobalProperties.PREFIX, Bindable.of(AzureGlobalProperties.class))
             .orElseGet(AzureGlobalProperties::new);
 
-        AzureKeyVaultSecretProperties existingValue = new AzureKeyVaultSecretProperties();
-        AzureGlobalPropertiesUtils.loadProperties(azureProperties, existingValue);
+        AzureKeyVaultSecretProperties secretProperties = binder
+                .bind(AzureKeyVaultSecretProperties.PREFIX, Bindable.of(AzureKeyVaultSecretProperties.class))
+                .orElseGet(AzureKeyVaultSecretProperties::new);
+
+        List<AzureKeyVaultPropertySourceProperties> list = secretProperties.getPropertySources();
+
+        // Load properties from global properties.
+        for (int i = 0; i < list.size(); i++) {
+            list.set(i, buildMergedProperties(globalProperties, list.get(i)));
+        }
```

Comment:
How about using another list instead of changing the list directly.

Reply:
I think both is OK: 
1. Set the whole list to a new list.
2. Set the list item one by one.

Neither of the 2 approach can not keep a immutable `AzureKeyVaultSecretProperties`.